### PR TITLE
Create separate RELEASE.md file, update release instructions

### LINKED ISSUE #2013: Move release instructions from README into a seperate file
cc @bhousel 

ref https://github.com/mapbox/mapbox-gl-js/issues/2000

### PR DESCRIPTION
**Pull request by lucaswoj**
fixes #2013 

cc @bhousel

### CODE REVIEW

**Comment by lucaswoj** [comment]
- [ ] create easier `npm run start-docs` script
 - [ ] add a step to check for release blockers

**Comment by lucaswoj** [comment]
This is ready for :eyes: @mourner @jfirebaugh @ansis @bhousel

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+```bash
+git checkout mb-pages
+git merge master
+git push origin mb-pages
+git checkout master
+```
+
+## Publish to npm
+
+Publish the release to npm. **There is no going back once you execute this command! A published npm package is forever.**
```
Comment (jfirebaugh) [inline RELEASE.md]: When the CI build publishes to the CDN, that's forever too.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# Release Procedure
+
+## Check for "Release Blocker" issues or PRs
+
+If there are any [open "Release Blocker" issues or PRs](https://github.com/mapbox/mapbox-gl-js/labels/release%20blocker), we cannot do a release.
+
+## Merge `mb-pages` into `master`
+
+```bash
+git checkout master
```
Comment (jfirebaugh) [inline RELEASE.md]: We've been implying a `git fetch origin` here -- let's make it explicit.

**Comment by jfirebaugh** [comment]
:+1:

File: RELEASE.md (hunk 1 above)
Comment (lucaswoj) [inline RELEASE.md]: Are you sure? I was under the impression the CDN hosted versions would be overwritten (eventually) if you deleted and re-pushed the tag.

Comment (jfirebaugh) [inline RELEASE.md]: They will get overwritten on S3, but not in browser or mid-tier caches. We set a really long expiration time on static assets.

Comment (lucaswoj) [inline RELEASE.md]: :+1: I'll add a scary warning to that line too.

**Pull request merged**